submissions/kikinovk/Memory_Pair-game/js/main.js

```diff
@@ -0,0 +1,123 @@
+const images = [
+  { src : './img/card_1.svg',
+    id : '01'
+  },
+  { src : './img/card_2.svg',
+    id : '02'
+  },
+  { src : './img/card_3.svg',
+      id : '03'
+  },
+  { src : './img/card_4.svg',
+    id : '04'
+  },
+  { src : './img/card_5.svg',
+    id : '05'
+  },
+  { src : './img/card_6.svg',
+    id : '06'
+  }
+];
+
+let openCard;
+const delay = 500;
+const maxCard = 6;
+const cards = [];
+const gameBoard = document.querySelector('.gameboard');
+const restartButton = document.querySelector('#restart_game');
+
+const selectImage = (cards, images) => {
+  let maxRandom = images.length;
+  let random = 0;
+
+  do {
+    random = Math.floor(Math.random() * Math.floor(maxRandom));
+  } while (cards
+              .filter(card => card.dataset.id === images[random].id)
+              .length === 2);
+  return images[random]
+};
+
+const creatCard = ({src, id}) => {
+  let card =  document.createElement('div');
+
+  card.classList.add('card');
+  card.dataset.id = id;
+  card.innerHTML = `<div class="card__front">
+                      <img src="./img/face.png" class="card__img--front" >
+                    </div>
+                    <div class="card__back">
+                      <img src="${src}" class="card__img--back">
+                    </div>`;
+  return card
+};
+
+const initGameBoard = () => {
+  const fragment = document.createDocumentFragment();
+
+  for (let i = 0; i < maxCard*2; i++) {
+    let card = creatCard(selectImage(cards, images));
+
+    fragment.appendChild(card);
+    cards.push(card);
+  };
+  gameBoard.appendChild(fragment);
+};
+
+const openedCard = (card) => {
+  card.classList.add('card--open');
+};
+
+const closedCard = (card) => {
+  if (card.classList.contains('card--open')) {
+    setTimeout(() => {
+      card.classList.remove('card--open')
+    }, delay);
+    card.open = false;
+  };
+};
+
+const disappearCard = (card) => {
+  setTimeout(() => {
+    card.classList.add('card--disappear')
+    setTimeout(() => {
+      card.classList.add('card--disable')
+    }, delay);
+  },  delay*2);
+};
+
+const eventCard = () => {
+  cards.forEach(card => {
+    card.addEventListener('click', () => {
+      openedCard(card);
+      if (!openCard) {
+       openCard = card;
+       card.open = true;
```

Comment:
are you sure that your card have `open` property ?

Reply:
The `open` property is declared for the first time in this part of the code. I can't do that?

Reply:
Is it better to declare the `open` property in the `creatCard` function?

Reply:
what stored in `card` variable ?

Reply:
in `card` variable  the card itself from the `Cards` array
![image](https://user-images.githubusercontent.com/9951988/108773708-6e725500-7567-11eb-85a0-12c41d0d4366.png)

Reply:
![image](https://user-images.githubusercontent.com/9951988/108774196-05d7a800-7568-11eb-8297-603406af4bb6.png)

Reply:
It is better to not mix render data (html) with properties/state and so on.

Reply:
[Separation_of_concerns](https://en.wikipedia.org/wiki/Separation_of_concerns)

Reply:
can I use `cards.classList.contains('.card--open') === true` instead of `card.open === true`?

Reply:
yep